lib/sshutils/sftp/sftp_test.go

```diff
@@ -583,6 +587,83 @@ func TestCopyingSymlinkedFile(t *testing.T) {
 	checkTransfer(t, false, dstPath, linkPath)
 }
 
+func TestHTTPUpload(t *testing.T) {
+	tempDir := t.TempDir()
+	src := "source"
+	dst := filepath.Join(tempDir, "destination")
+
+	createFile(t, tempDir, src)
+	src = filepath.Join(tempDir, src)
```

Comment:
It's a little weird for src to start out with one value and then be overwritten. It doesn't seem necessary here.

Reply:
Now that you mention it that's true. I changed `createFile` and `createDir` to take a single path so I could clean instances like this up.